### PR TITLE
Feature/xtg 289 - Capabilities System

### PR DESCRIPTION
**Pull request by fredericobraga85**
ticke: https://x-team-internal.atlassian.net/browse/XTG-289

created Capability table
added default capabilities for each Role
added capabilities logic in endpoints and getAuthMiddleware
added tests for routes and user

Note: about the r size, the majority of line changes are in tests  ;) Trying to add as much tests as possible so that we can reach a solid testing env in a few weeks

### CODE REVIEW

File: src/api-utils/midddleware/getAuthUser.ts [hunk 1]
```diff
@@ -21,16 +21,26 @@ export async function getAuthUser(
...
-  // return h.continue;
+}
+
+function userRoleContainsCapability(
+  requiredCapabilities: string[],
+  userCapabilities?: Capability[]
+): boolean {
+  if (!requiredCapabilities.length) {
+    return true;
+  }
```
Comment (fredericobraga85) [inline src/api-utils/midddleware/getAuthUser.ts:41]: endpoints with requiredCapabilities are defaulted to allow access

File: src/games/utils/index.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export function adminAction(userRequesting: User): boolean {
   return !!(
     userRequesting.isSuperAdmin() ||
     userRequesting.isAdmin() ||
-    userRequesting.isCommunityTeam()
+    userRequesting.isGameDev()
```
Comment (fredericobraga85) [inline src/games/utils/index.ts:70]: I exepct to completey remove this functions , but will leave it for another pr since it will require a deeper look into Slack commands

**Review by ramirlm — APPROVED** [review]
Nice stuff! Makes sense to me 👍

**Review by ccmoralesj — APPROVED** [review]
Quite a huge work. This is a before and after for this API. The capabilities system will help us a lot. Great work @fredericobraga85 !!
